src/Microsoft.AspNetCore.SpaTemplates/.gitignore

```diff
@@ -0,0 +1,2 @@
+content/*/*.csproj
+content/*/*.fsproj
```

Comment:
This system is a bit awkward because it looks like it prevents you from having a `.sln` that includes the template `.csproj`es. You'd need to be careful to run the `build` task to generate the `csproj` files before you could open the `.sln`.

Are you still considering whether you could make the original `csproj` files runnable directly like they were before?

Reply:
I played around with options, but didn't find anything that worked well. The difficulty was in differentiating between MSBuild variables and template variables.  My primary concern was to remove the system in which we have to edit files after they are placed in the nupkg. If you have other ideas for a way to keep you existing workflow, let me know.

Reply:
What's so problematic about doing the substitution after the nupkg is created?

Reply:
> it looks like it prevents you from having a .sln that includes the template .csprojes

We could still have a .sln file for this. You would just have to execute `dotnet build` on `Templating.sln` first to generate the csproj files.

Reply:
OK, understood. At the same time, this approach breaks other basic scenarios like opening a solution in VS (without first doing a special action). A key idea of the `dotnet new` project system is [runnable projects](https://github.com/dotnet/templating/wiki/%22Runnable-Project%22-Templates) that are usable directly from source.

What you're adding here is totally fine for the `rel/2.0.x` branch, and thanks for sorting it out so quickly.

When we do this for real in `dev`, I'd love it if we could find a cleaner solution that keeps the build pipeline working in the expected ways, and without forcing devs to go through any unexpected extra steps. One possible approach would be *not* to replace these tokens at all during the package creation, so the resulting `.nupkg` contains `.csproj` files with dependencies expressed like this:

```
<PackageReference Include="Microsoft.AspNetCore.All" Version="$(TemplateDep_Coherence_2_1)" />
```

This would be runnable directly from source (and could be included in a regular `.sln`, etc.), because a `Directory.Build.props` file (or similar) would supply values based on the new `build/dependencies.props` you've added.

Then, when someone executes `dotnet new <template>`, we could have `dotnet new`'s template system resolve the symbols to fixed numbers. To do this we'd need entries in the `.template.config/template.json` files, adding to the `symbols` settings entries like `"TemplateDep_Coherence_2_1": { "replaces": "$(TemplateDep_Coherence_2_1)", "value": "2.1.2" }`.

So to make this work, we'd want our build system to generate or augment the `.template.config/template.json` files during the build. Previously you said it might be possible to modify the `.csproj` files in the staging directory during the build, so perhaps the same technique could be used instead to modify the `.template.config/template.json` files.

What do you think? This is just one possibility. If you have any other ideas that's great too!

Reply:
oooh, i like this. That would work well too. Didn't know this was a thing the templating system supported. Does the `.template.config/template.json` + `replaces` system work today? And btw, is there some documentation on available settings in template.json ?

Reply:
Yes, it is supported today.

Docs for dotnet new's substitution features are all at [the link from before](https://github.com/dotnet/templating/wiki/%22Runnable-Project%22-Templates), or @sayedihashimi would know everything there is to know about it.